Tavisca.Bootcamp.LanguageBasics.Exercise1/Program.cs

```diff
@@ -23,61 +23,57 @@ private static void Test(string args, int expected)
         public static int FindDigit(string equation)
         {
             String[] expressions = equation.Split(new Char[] { '*', '=' });        //here we split the equation into three parts
-            int[] initialize = new int[3];                                         //here we made an array for basically storing at which part is an '?'
-            for (int i = 0; i < initialize.Length; i++)
-            {
-                initialize[i] = -1;                                                 //initially we initialzie the array by -1 
-            }
-            initialize = checkExpression(initialize,expressions);
-            int expression1 = initialize[0];
-            int expression2 = initialize[1];
-            int expression3 = initialize[2];
-            int ans = 0;
+           
+            int expression1=CheckForQuestionMark(expressions[0]); 
+            int expression2=CheckForQuestionMark(expressions[1]); 
+            int expression3=CheckForQuestionMark(expressions[2]);
+
+            int finalAns = 0;
             if (expression3 == -1)
             {                                                                             //if num3 contains '?' than we have to multilpy
-                ans = multiply1(expression1, expression2, expressions[2]);
+                int multipliedExpression=expression1*expression2;
+                finalAns = MultipliedResult(multipliedExpression, expressions[2]);
             }
             else if (expression1 == -1)                                                         //if num1 contains '?' than we have to divide
             {
-                ans = divide1(expression2, expression3, expressions[0]);
+                int divideExpression = expression3/expression2;
+                if(expression3%divideExpression != 0)
+                {
+                    return -1;
+                }
+                finalAns = DivideResult(divideExpression, expressions[0]);
             }
             else if (expression2 == -1)                                                          //if num2 contains '?' than we have to divide
             {
-                ans = divide1(expression1, expression3, expressions[1]);
+                int divideExpression = expression3/expression1;

```

Comment:
'divideExpression' is not an expression, it's the result of dividing the two numbers. 
You need to checkup different parts of the equation and rename the variables accordingly.

In this problem A and B are operands, * is the operator and C is the result.